### ISSUE TITLE
Need help writing some midje tests for an error-function 

### ISSUE THREAD

**Issue by Vaguery** [lspector/Clojush#146]
I'd like to write the requisite tests for my error function in `problems/tozier/winkler01.clj` but I have only very sketchy ideas how to gather together the arguments and exercise functions elsewhere in the code.

So the basic approach for any given `fact` seems like:

1. load my file with the `error-function` defined in it
2. create a `program` with a plannable outcome; for example, one that produces a 0 error score, one that produces a positive error score, one that doesn't give an answer, one that's empty
3. apply the `error-function` to the `program` and compare the result to the expected result

Problem is, I can't suss out:

1. What do I do to create the argument for an `error-function`?
1. What is the structure returned by a typical `error-function`?
1. What do you "normally" do when a calculation might throw an exception? For example, in `winkler01` I would expect the function `proportion-not-01` to throw a `Divide by 0` error if it were ever handed a non-numeric argument. So what's the _idiom_ for handling that in the code, and what's your codebase's _standard_? (not necessarily the same thing)

**Comment by thelmuth** [comment]
I don't know if you've figured any of this out on your own yet, and I don't understand all of your questions, but maybe this will help:

On 2.: Every error function in Clojush is expected to return a sequence (i.e. list or vector) of errors, all of which should be `>= 0`. That's the only requirement on error function returns.

On 3.: If I'm understanding you correctly, then we just let Clojure/Java's exception handling take care of the exceptions. If something divides by zero, or tries to `(+ "abc" "xyz")`, Clojure will raise an exception and everything will crash. We don't do anything to catch exceptions.

**Comment by Vaguery** [comment]
Actually I figured out some of what I wanted to know, and it's in PR #145. For the record:

1. the argument for an `error-function` is a _program_, and what wasn't immediately clear from the code was how to _make_ a program. It's just a quoted list of Push tokens, which I finally worked out by trial and error. So `'( 1 false integer_add )` works as the argument.
2. Quick followup on error values: if they _are_ (by chance or mistake) negative, what will break?
3. right; so what you'er saying is that exception handling should be an explicit part of the functions we write.
